gcc/rust/resolve/rust-ast-resolve-pattern.h

```diff
@@ -25,6 +25,14 @@
 namespace Rust {
 namespace Resolver {
 
+enum PatternBoundCtx
+{
+  Product,
+  Or,
+};
+
+using PatternBinding = std::pair<PatternBoundCtx, std::set<Identifier>>;
```

Comment:
This seems more like a type-def than a using. Please avoid using statements. Or make this your own struct